agent.go

```diff
@@ -985,9 +987,30 @@ func (a *Agent) sendBindingSuccess(m *stun.Message, local, remote Candidate) {
 	}
 }
 
+// Removes pending binding requests that are over maxBindingRequestTimeout old
+func (a *Agent) invalidatePendingBindingRequests() int {
+	nRemoved := 0
+	temp := a.pendingBindingRequests[:0]
+	for _, bindingRequest := range a.pendingBindingRequests {
+		if time.Since(bindingRequest.timestamp) < maxBindingRequestTimeout {
+			temp = append(temp, bindingRequest)
+		} else {
+			nRemoved++
+		}
+	}
+	// Garbage collect removed elements
+	for i := len(temp); i < len(a.pendingBindingRequests); i++ {
+		a.pendingBindingRequests[i] = bindingRequest{}
+	}
+	a.pendingBindingRequests = temp
+	return nRemoved
```

Comment:
Instead of `nRemoved` what do you think of doing `preInvalidateLength - len(a.pendingBindingRequests )`?

Just nice to have a little less state, I read this function a few times to make sure I fully understood everything that was happening. If you disagree please push back, I don't feel strongly :)